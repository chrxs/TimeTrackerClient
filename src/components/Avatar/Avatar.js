import React from 'react'
import PropTypes from 'prop-types'

const Avatar = ({
  imageUrl,
  style,
  ...rest
}) => {
  style = {
    ...avatarStyle,
    backgroundImage: `url(${imageUrl})`,
    ...style
  }
  return (
    <div style={style} {...rest} />
  )
}

const avatarStyle = {
  backgroundPosition: '50% 50%',
  backgroundRepeat: 'no-repeat',
  backgroundColor: '#ccc',
  backgroundSize: 'cover',
  borderRadius: '1000px',
  overflow: 'hidden',
  margin: '0 auto',
  width: '50px',
  height: '50px'
}

Avatar.displayName = 'Avatar'

Avatar.propTypes = {
  imageUrl: PropTypes.string,
  style: PropTypes.object
}

Avatar.defaultProps = {
  style: {}
}

export default Avatar
