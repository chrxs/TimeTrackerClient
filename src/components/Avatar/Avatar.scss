.Avatar {
  background-position: 50% 50%;
  background-repeat: no-repeat;
  background-color: #ccc;
  background-size: cover;
  border-radius: 1000px;
  box-sizing: border-box;
  overflow: hidden;
  margin: 0 auto;
  width: 50px;
  height: 50px;
}
