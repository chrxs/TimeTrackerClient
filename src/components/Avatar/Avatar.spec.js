/* global describe, test, expect */
import React from 'react'
import Avatar from './Avatar'
import renderer from 'react-test-renderer'

describe('Avatar snapshots', () => {
  test('with no props', () => {
    const component = renderer.create(<Avatar />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with imageUrl prop', () => {
    const component = renderer.create(<Avatar imageUrl='http://image.jpg' />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
