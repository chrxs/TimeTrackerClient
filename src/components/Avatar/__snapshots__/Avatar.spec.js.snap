// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshots with imageUrl prop 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ccc",
      "backgroundImage": "url(http://image.jpg)",
      "backgroundPosition": "50% 50%",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "borderRadius": "1000px",
      "height": "50px",
      "margin": "0 auto",
      "overflow": "hidden",
      "width": "50px",
    }
  }
/>
`;

exports[`Avatar snapshots with no props 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ccc",
      "backgroundImage": "url(undefined)",
      "backgroundPosition": "50% 50%",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "borderRadius": "1000px",
      "height": "50px",
      "margin": "0 auto",
      "overflow": "hidden",
      "width": "50px",
    }
  }
/>
`;
